src/current/advisories/a131639.md

```diff
@@ -0,0 +1,211 @@
+
+While transactions will typically be routed from gateway nodes to the true leaseholder (on the node with the healthy store), and write intents will be written on this true leaseholder, on the node with the slow store these write intents are still queued up in the range's Raft log, and have not been applied. If async intent resolution (the means by which intents are committed to durable records, once a transaction commits) is subsequently triggered on this slow node before it has caught up on its Raft log, intent resolution becomes a no-op, instead of marking the intent as committed. A future transaction that interacts with the row may then come across this intent, and if the committed transaction's record has been garbage collected at that point, it may incorrectly determine that the intent belongs to an aborted transaction. This would then cause the intent to be cleaned up, effectively dropping the committed write.
+
+This bug is rare in that it requires intermittent slowness of a block device backing a store, coupled with lease transfers onto the store with the intermittently slow block device, followed by async intent resolution on the slow node before it has caught up on its Raft log. Writes must touch more than one range. Single range writes, including those that benefit from the One-Phase-Commit (1PC) fast path are not affected.
+
+All CockroachDB versions prior to and including the following are impacted by this bug:
+
+- v23.1.0 - v23.1.26 and below
+- v23.2.0 - v23.2.10 and below
+- v24.1.0 - subsequent patch versions are unaffected
+
+Versions prior to 23.1 are no longer eligible for [maintenance support]({% link releases/release-support-policy.md %}), and this issue will not be addressed in those versions.
+
+## Statement
+
+[#123442](https://github.com/cockroachdb/cockroach/pull/123442) resolves an issue in CockroachDB in the expiration-to-epoch lease promotion transition, where the lease's effective expiration could be allowed to regress, resulting in two nodes believing they are the leaseholder for a range.
+
+The patch has been applied to maintenance releases of CockroachDB:
+
+- [v23.1.27]({% link releases/v23.1.md%}#v23-1-27)
+- [v23.2.11]({% link releases/v23.2.md%}#v23-2-11)
+- [v24.1.1]({% link releases/v24.1.md%}#v24-1-1)
+
+This public issue is tracked by [131639](https://github.com/cockroachdb/cockroach/issues/131639).
+
+## Mitigation
+
+Users are encouraged to upgrade to [v23.1.27]({% link releases/v23.1.md%}#v23-1-27), [v23.2.11]({% link releases/v23.2.md%}#v23-2-11), [v24.1.1]({% link releases/v24.1.md%}#v24-1-1), or a later version that includes the patch.
+
+### Detection via logs
+
+Users can run the script included below to detect part of the symptoms of this technical advisory. The script accepts a [debug.zip]({% link v24.3/cockroach-debug-zip.md %}) and determines whether any lease expiration regressions are evident from the logs.
```

Comment:
```suggestion
Users can run the script [a131639_check.sh](a131639_check.sh) to determine whether any lease expiration regressions are evident from their logs. The script must be run from within a decompressed [debug.zip]({% link v24.3/cockroach-debug-zip.md %}) directory.
```

@nvanbenschoten My suggestion here (1) links to a copy of the script in the docs repo (2) implements your comment (3) removes the language that I found too vague that the script will "detect part of the symptoms of this technical advisory".

I presume users would benefit from a more specific caveat about what the script may or may not find, but LMK what you think, or if this is OK as-is.